terraso_backend/config/urls.py

```diff
@@ -13,13 +13,65 @@
 # You should have received a copy of the GNU Affero General Public License
 # along with this program. If not, see https://www.gnu.org/licenses/.
 
+import oauth2_provider.views as oauth2_views
+from django.conf import settings
 from django.contrib import admin
 from django.urls import include, path
 
+# OAuth2 provider endpoints
+oauth2_endpoint_views = [
+    path("authorize/", oauth2_views.AuthorizationView.as_view(), name="authorize"),
+    path("token/", oauth2_views.TokenView.as_view(), name="token"),
+    path("revoke-token/", oauth2_views.RevokeTokenView.as_view(), name="revoke-token"),
+]
+
+if settings.OAUTH2_PROVIDER.get("OIDC_ENABLED", False):
+    # OIDC specific views
+    oauth2_endpoint_views += [
+        path(
+            ".well-known/openid-configuration/",
+            oauth2_views.ConnectDiscoveryInfoView.as_view(),
+            name="openid-configuration",
+        ),
+        path(".well-known/jwks.json", oauth2_views.JwksInfoView.as_view(), name="jwks-info"),
+        path("userinfo/", oauth2_views.UserInfoView.as_view(), name="user-info"),
+    ]
+
+if settings.DEBUG:
+    # OAuth2 Application Management endpoints
+    oauth2_endpoint_views += [
+        path("applications/", oauth2_views.ApplicationList.as_view(), name="list"),
+        path(
+            "applications/register/",
+            oauth2_views.ApplicationRegistration.as_view(),
+            name="register",
+        ),
+        path("applications/<pk>/", oauth2_views.ApplicationDetail.as_view(), name="detail"),
+        path("applications/<pk>/delete/", oauth2_views.ApplicationDelete.as_view(), name="delete"),
+        path("applications/<pk>/update/", oauth2_views.ApplicationUpdate.as_view(), name="update"),
+    ]
+
+    # OAuth2 Token Management endpoints
+    oauth2_endpoint_views += [
+        path(
+            "authorized-tokens/",
+            oauth2_views.AuthorizedTokensListView.as_view(),
+            name="authorized-token-list",
+        ),
+        path(
+            "authorized-tokens/<pk>/delete/",
+            oauth2_views.AuthorizedTokenDeleteView.as_view(),
+            name="authorized-token-delete",
+        ),
+    ]
+
+
```

Comment:
question: Are we manually re-adding the paths we removed via `oauth2_provider.urls`? If so, is there a less duplicative way of doing this?

Reply:
Yeah, the idea is to remove certain paths that allow any authenticated user to add new OAuth applications to Terraso. I basically copied an [example](https://django-oauth-toolkit.readthedocs.io/en/latest/tutorial/tutorial_02.html) from the django-oauth-toolkit docs, so unfortunately I don't think there is an easier way.